### PR TITLE
moq-relay: init at v0.6.2

### PR DESCRIPTION
**Pull request by therishidesai**
## Description of changes

Create nixos module for moq-relay. moq-relay package comes from #342588

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by s0me1newithhand7s** [comment]
pls, try to not ping 3500+ ppls this time. 💀

**Comment by hraban** [comment]
> pls, try to not ping 3500+ ppls this time. 💀

🤷 as one of the people getting pinged, at what point do we take responsibility for the fragility of this system? do we truly expect nobody to ever make the mistake of merging off staging and accidentally creating a PR into the _default_ base branch offered by GH (master)? 😛 I know I've been there.

thanks for the contribution  @therishidesai  :)

**Comment by therishidesai** [comment]
> pls, try to not ping 3500+ ppls this time. 💀

Sorry about that

**Comment by nyabinary** [comment]
Can't you just have one PR and init the package in one commit and init the module in the other 2 commits?

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5828,6 +5828,10 @@ with pkgs;
 
   mousetweaks = callPackage ../applications/accessibility/mousetweaks { };
 
+  moq-relay = callPackage ../by-name/mo/moq-relay/package.nix {
+    inherit libressl pkg-config;
+  };
+
```
Comment (nyabinary) [inline pkgs/top-level/all-packages.nix:5834]: IIRC, this isn't required for packages in ``by-name``.

**Review by nyabinary — CHANGES_REQUESTED** [review]
IIRC this isn't required for packages in ``by-name``.

File: pkgs/by-name/mo/moq-relay/package.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    hash = "sha256-J9f+opePgvtsi7LfK7cH0kScB5xfZttY0Nrg89cypMs=";
+  };
+
+  cargoHash = "sha256-34bvlV7IJ0bn6Ghr6qH8Yh08xt2v17Mnj2Ej0h1pHM4=";
+
+  nativeBuildInputs = [ pkg-config ];
+
+  buildInputs = [ libressl ];
+
+  meta = with lib; {
```
Comment (dotlambda) [inline pkgs/by-name/mo/moq-relay/package.nix:27]:
```suggestion
  meta = {
```
#208242

File: nixos/modules/services/video/moq-relay.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      type = lib.types.str;
+      description = "Path to TLS key";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.moq-relay = {
+      description = "Media over QUIC relay server";
+      path = [ pkgs.moq-relay ];
+      script = ''
```
Comment (dotlambda) [inline nixos/modules/services/video/moq-relay.nix:32]: use `ExecStart` instead

**Review by dotlambda — COMMENTED** [review]
Commit message should be `moq-relay: init at 0.6.0`.
Please add some hardening to the service.

**Comment by therishidesai** [comment]
> Can't you just have one PR and init the package in one commit and init the module in the other 2 commits?

Was just basing this off of other packages where the package was added in an init PR and the nixos module was in another PR

**Comment by therishidesai** [comment]
> Commit message should be `moq-relay: init at 0.6.0`. Please add some hardening to the service.

I had a separate PR to just add the `moq-relay` package. Is it fine that this PR has 2 commits? In the past I remember maintainers asking for single atomic commits in the PR

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (therishidesai) [reply pkgs/top-level/all-packages.nix]: fixed

**Comment by therishidesai** [comment]
> Please add some hardening to the service.

I'm not too familiar with the systemd hardening. Are there any services I should look at for hardening examples? What specifically makes sense for this server?

**Comment by dotlambda** [comment]
> Are there any services I should look at for hardening examples?

Just grep for "hardening" in `nixos/`.

**Comment by therishidesai** [comment]
> > Are there any services I should look at for hardening examples?
> 
> Just grep for "hardening" in `nixos/`. Also see https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html.

Added some new hardening and cleaned up users/groups similar to other server modules in nixos.

**Pull request closed without merging**